sql/analyzer/resolve_views.go

```diff
@@ -109,10 +109,53 @@ func resolveViews(ctx *sql.Context, a *Analyzer, n sql.Node, scope *Scope, sel R
 	})
 }
 
+// applyAsOfToView transforms the nodes in the view's execution plan to apply the asOf expression to every
+// individual table involved in the view.
 func applyAsOfToView(n sql.Node, a *Analyzer, asOf sql.Expression) (sql.Node, transform.TreeIdentity, error) {
 	a.Log("applying AS OF clause to view definition")
 
-	return transform.Node(n, func(n sql.Node) (sql.Node, transform.TreeIdentity, error) {
+	// Transform any tables in our node tree so that they use the AsOf expression
+	newNode, nodeIdentity, err := applyAsOfToViewTables(n, a, asOf)
+	if err != nil {
+		return n, transform.SameTree, err
+	}
+
+	// Subquery expressions won't get updated by the Node transform above, but we still need to update
+	// any UnresolvedTable references in them to set the AsOf expression correctly.
+	newNode, exprIdentity, err := applyAsOfToViewSubqueries(newNode, a, asOf)
```

Comment:
I had to introduce a second transform to get to the tables nested inside subquery expressions. Another option would be to introduce a `NodeDelegate` interface (or something similar) that `Subquery` implements and then have a new transform method that traverses Opaque nodes and looks for expressions that are NodeDelegates and pull out their nodes for traversal. 

It seemed simpler and smaller scope to just do the two transforms here, so that was my preference, but if you think that would be a useful transform API, I'm happy to see what that looks like.

Reply:
No I think this is fine